core/commonMain/src/implementations/persistentOrderedMap/PersistentOrderedMapBuilder.kt

```diff
@@ -23,15 +25,15 @@ internal class PersistentOrderedMapBuilder<K, V>(private var map: PersistentOrde
     override val size: Int get() = hashMapBuilder.size
 
     override fun build(): PersistentMap<K, V> {
-        val newHashMap = hashMapBuilder.build()
-        map = if (newHashMap === map.hashMap) {
+        return builtMap?.also { map ->
```

Comment:
Given a more complex logic of updating the `builtMap`,  WDYT about adding the following consistency checks?
* `check(hashMapBuilder.builtMap != null)` when `builtMap` is not null, and
* `check(hashMapBuilder.builtMap == null)` when `builtMap` is null

This would require `hashMapBuilder.builtMap` be `internal`, but with `private set`.

Reply:
And the same for `PersistentOrderedSetBuilder`.

Reply:
done!